autorag/nodes/queryexpansion/run.py

```diff
@@ -0,0 +1,174 @@
+import logging
+import os
+import pathlib
+from typing import List, Callable, Dict, Tuple
+
+import pandas as pd
+
+from autorag import embedding_models
+from autorag.nodes.retrieval import bm25, vectordb
+from autorag.nodes.retrieval.base import load_bm25_corpus, load_chroma_collection
+from autorag.nodes.retrieval.run import evaluate_retrieval_node
+from autorag.schema import Module
+from autorag.strategy import measure_speed, filter_by_threshold, select_best_average
+from autorag.utils.util import make_module_file_name, fetch_contents, make_combinations
+
+logger = logging.getLogger("AutoRAG")
+
+
+def run_query_expansion_node(modules: List[Callable],
+                             module_params: List[Dict],
+                             previous_result: pd.DataFrame,
+                             node_line_dir: str,
+                             strategies: Dict,
+                             ) -> pd.DataFrame:
+    """
+    Run evaluation and select the best module among query expansion node results.
+
+    :param modules: Query expansion modules to run.
+    :param module_params: Query expansion module parameters.
+    :param previous_result: Previous result dataframe.
+        In this case, it would be qa data.
+    :param node_line_dir: This node line's directory.
+    :param strategies: Strategies for query expansion node.
+    :return: The best result dataframe.
+    """
+    if not os.path.exists(node_line_dir):
+        os.makedirs(node_line_dir)
+    project_dir = pathlib.PurePath(node_line_dir).parent.parent
+    resources_dir = os.path.join(project_dir, "resources")
+    data_dir = os.path.join(project_dir, "data")
+    retrieval_gt = pd.read_parquet(os.path.join(project_dir, "data", "qa.parquet"))['retrieval_gt'].tolist()
+
+    # run query expansion
+    result_queries, execution_times = zip(*map(lambda task: measure_speed(
+        task[0], project_dir=project_dir, previous_result=previous_result, **task[1]), zip(modules, module_params)))
+    average_times = list(map(lambda x: x / len(result_queries[0]), execution_times))
+
+    # pop top_k from strategies
+    if "top_k" in strategies.keys():
+        top_k = strategies.pop("top_k")
+    else:
+        top_k = 10  # default value
```

Comment:
There will be other node parameters.
Do not treat 'top_k' as special parameter, treat as node parameter.
You can pop other keys and treat left keys as node parameters.